cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -319,18 +321,37 @@ def _add_alarms(self):
             ),
         }
 
+        if self._condition_is_slurm() and is_feature_supported(Feature.CLUSTER_HEALTH_METRICS):
+            metrics_for_alarms["ProtectedMode"] = cloudwatch.Metric(
+                namespace="ParallelCluster",
+                metric_name="ClusterInProtectedMode",
+                dimensions_map={"ClusterName": self.stack.stack_name},
+                statistic="SampleCount",
+                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
+            )
+
         for metric_key, metric in metrics_for_alarms.items():
             alarm_id = f"HeadNode{metric_key}Alarm"
             alarm_name = f"{self.stack.stack_name}_{metric_key}Alarm_HeadNode"
+
+            threshold_value = (
```

Comment:
To make the code cleaner and extendable, could you instead modify the metrics_for_alarm dict to extend threshod_value and comparison_operator there?
For example
```
metrics_for_alarms = {
        "Mem": {
            "metric": cloudwatch.Metric(
                namespace="CWAgent",
                metric_name="mem_used_percent",
                dimensions_map={"InstanceId": self.head_node_instance.ref},
                statistic="Maximum",
                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
            ),
            "threshold": CW_ALARM_PERCENT_THRESHOLD_DEFAULT,
            "comparison_operator": cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
        },
```

Reply:
Added a new dictionary called alarm_settings!